bundles/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java

```diff
@@ -174,6 +174,17 @@ public void call(Object... args) {
                     }
                 });
             }
+        }).on(Manager.EVENT_CONNECT_ERROR, new Emitter.Listener() {
+            @Override
+            public void call(Object... args) {
+                // Try reconnecting on connection errors
+                if (logger.isDebugEnabled()) {
+                    logger.error("Error connecting to the openHAB Cloud instance: {}. Reconnecting.", args[0]);
+                } else {
+                    logger.error("Error connecting to the openHAB Cloud instance. Reconnecting.");
+                }
+                socket.close().connect();
```

Comment:
To be checked: throttling/delay of retries

Reply:
I agree, it would be nice to both delay (maybe randomize it) and throttle this call.  One issue we have with the cloud service is the "Thundering Hurd" problem where when we disconnect a large number of users (loose a load balancer, or more likely linode has a network issue) , those  users constantly DDOS us and try to reconnect at the same time.  In the case of a total network outage, this can be 20k-30k clients.